### DISCUSSION TITLE
Expose provider type on OAuthAccountNotLinked error

### DISCUSSION THREAD

**Discussion by emperorgurgeh** [nextauthjs/next-auth#3670] [Ideas]
### Description 📓

Right now, if a user tries to sign in with a different provider but same email as they previously had, nextauth throws an OAuthAccountNotLinkedError

With this we can tell our users "An account already exists with this email and another sign-in method"

If the error also exposed the original provider method, we could make this error 10x better by telling them "You previously signed in with Facebook and should sign in with it again"

I'm 99% certain this would not introduce any security issue

### How to reproduce ☕️

N/A

### Contributing 🙌🏽

Yes, I am willing to help implement this feature in a PR

**Comment by ghost** [comment]
I'm also struggling with this issue. 

Not linking the same email, What would be the best way to handle this flow? 

Using signIn callback in [...nextauth].js?
